src/core/public/ui_settings/ui_settings_service.test.ts

```diff
@@ -34,7 +34,10 @@ describe('#stop', () => {
     service.stop();
 
     await expect(
-      Rx.combineLatest(client.getUpdate$(), client.getUpdateErrors$(), loadingCount$!).toPromise()
+      Rx.lastValueFrom(
+        Rx.combineLatest([client.getUpdate$(), client.getUpdateErrors$(), loadingCount$!]),
+        { defaultValue: undefined }
+      )
```

Comment:
This is not strictly the same thing, is it?

Reply:
The way I understand the previous code
```
await expect(
  Rx.combineLatest(client.getUpdate$(), client.getUpdateErrors$(), loadingCount$!).toPromise()
).resolves.toBe(undefined);
```
is that we are expecting the observables to complete before emitting any value (we are not piping and taking any number of events, so essentially `toPromise()` will hold until the last value is emitted or the observables are complete).

However, the new `firstValueFrom` and `lastValueFrom` throw when the observable completes before emitting any value. `toPromise()` would simply return `undefined` in that case. 

To avoid `lastValueFrom` from throwing, you can provide a default value to fallback in case the observable closes before emitting.

Am I missing any other side effect the previous code had over this one?